store/postgres/src/store.rs

```diff
@@ -698,9 +703,26 @@ impl Store {
         Ok(storage.clone())
     }
 
-    fn cached_schema(&self, subgraph_id: &SubgraphDeploymentId) -> Result<SchemaPair, Error> {
-        if let Some(pair) = self.schema_cache.lock().unwrap().get(&subgraph_id) {
-            return Ok(pair.clone());
+    fn cached_schema(&self, subgraph_id: &SubgraphDeploymentId) -> Result<SchemaCacheEntry, Error> {
+        if let Some(entry) = self.schema_cache.lock().unwrap().get(&subgraph_id) {
+            // Cache entry is stale and an unstable import has changed or is now available
+            // TODO: Is it okay to ignore unstable imports which are no longer available?
+            // TODO: Make this threshold an envvar?
+            let requires_refresh = entry.last_refresh.elapsed().as_secs() >= 120
+                && entry
+                    .unstable_imports
+                    .iter()
+                    .any(|(schema_reference, subgraph_id_opt)| {
+                        self.resolve_schema_reference(schema_reference)
+                            .map(|schema| match subgraph_id_opt {
+                                Some(subgraph_id) => schema.id.eq(subgraph_id),
+                                None => true,
+                            })
+                            .unwrap_or(false)
+                    });
```

Comment:
Don't we also have to check the previously available schemas that were imported? Some of them might be gone now.